### PR TITLE
FEAT: automatically build docs upon merge to main

### PR DESCRIPTION
**Pull request by MikeLuDev**
- Adds workflow to build docs when changes are merged into main instead of us manually rebuilding them

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Docs publish
+
+# This will be on merge to main instead
```
Comment (turnyay) [inline .github/workflows/docs.yml:3]: old comment?

**Review by turnyay — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (MikeLuDev) [reply .github/workflows/docs.yml:3]: Old comment :stuck_out_tongue:

Comment (MikeLuDev) [inline .github/workflows/docs.yml:3]:
```suggestion
```

**Review by MikeLuDev — COMMENTED** [review]
tada

**Pull request merged**